File: aidl_client/src/main/java/com/phone/aidl_client/adapter/BookAdapter.java
package com.phone.aidl_client.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.phone.aidl_client.Book;
import com.phone.aidl_client.R;

import java.util.ArrayList;
import java.util.List;

/**
 * author : Urasaki
 * e-mail : 1164688204@qq.com
 * date   : 2021/6/28 13:34
 * desc   :
 * version: 1.0
 */
public class BookAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private static final String TAG = BookAdapter.class.getSimpleName();
    private Context mContext;
    private List<Book> mBookList = new ArrayList<>();

    public BookAdapter(Context context) {
        mContext = context;
    }

    public void clearData() {
        notifyItemRangeRemoved(0, this.mBookList.size());
        notifyItemRangeChanged(0, this.mBookList.size());
        this.mBookList.clear();
    }

    public void addData(List<Book> bookList) {
        notifyItemRangeInserted(mBookList.size(), bookList.size());
        mBookList.addAll(bookList);
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.item_book, parent, false);
        return new BodyHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        if (holder instanceof BodyHolder) {
            BodyHolder bodyHolder = (BodyHolder) holder;
            bodyHolder.tevBookName.setText(mBookList.get(position).getName());
            bodyHolder.tevBookContent.setText(mBookList.get(position).getContent());
        }
    }

    @Override
    public int getItemCount() {
        return mBookList.size();
    }

    private static class BodyHolder extends RecyclerView.ViewHolder {

        private TextView tevBookName;
        private TextView tevBookContent;

        public BodyHolder(@NonNull View itemView) {
            super(itemView);

            tevBookName = (TextView) itemView.findViewById(R.id.tev_book_name);
            tevBookContent = (TextView) itemView.findViewById(R.id.tev_book_content);
        }
    }
}

File: module_square/src/main/java/com/phone/module_square/adapter/BookAdapter.kt
package com.phone.module_square.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.phone.library_room.Book
import com.phone.module_square.R
import com.phone.module_square.databinding.SquareItemBookBinding


class BookAdapter(val context: Context, val bookList: MutableList<Book>) :
    RecyclerView.Adapter<ViewHolder>() {

    fun clearData() {
        notifyItemRangeRemoved(0, this.bookList.size)
        notifyItemRangeChanged(0, this.bookList.size)
        this.bookList.clear()
    }

    fun addData(bookList: MutableList<Book>) {
        notifyItemRangeInserted(this.bookList.size, bookList.size)
        this.bookList.addAll(bookList)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding: SquareItemBookBinding = DataBindingUtil.inflate(
            LayoutInflater.from(context), R.layout.square_item_book, parent, false
        )
        return ContentHolder(binding.root)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        //获取ViewDataBinding
        val binding = DataBindingUtil.getBinding<SquareItemBookBinding>(holder.itemView)?.apply {
            dataBean = bookList.get(position)
        }
        binding?.executePendingBindings()
    }

    override fun getItemCount(): Int {
        return bookList.size
    }

    private class ContentHolder(itemView: View) : RecyclerView.ViewHolder(itemView)


}